smart_card_connector_app/src/window-devices-displaying.js

```diff
@@ -92,65 +98,18 @@ function addDeviceClickListener(e) {
 /**
  * @param {!Array.<!chrome.usb.Device>} devices
  */
-function getDevicesCallback(devices) {
-  goog.array.sortByKey(devices, function(device) { return device.device; });
-
-  logger.info(devices.length + ' USB device(s) available: ' +
-              GSC.DebugDump.dump(devices));
-  displayReaders(devices);
-}
-
-/**
- * @param {!Array.<!chrome.usb.Device>} readers
- */
-function displayReaders(readers) {
-  goog.dom.removeChildren(readersListElement);
-  goog.array.forEach(readers, function(reader) {
-    GSC.Logging.checkWithLogger(logger, !goog.isNull(readersListElement));
-    goog.asserts.assert(readersListElement);
-    goog.dom.append(
-        readersListElement,
-        goog.dom.createDom(
-            'li', undefined, makeReaderDescriptionString(reader)));
-  });
-}
-
-/**
- * @param {!chrome.usb.Device} reader
- * @return {string}
- */
-function makeReaderDescriptionString(reader) {
-  var parts = [];
-
-  if (reader.productName) {
-    parts.push(chrome.i18n.getMessage(
-        'readerDescriptionProductNamePart', reader.productName));
-  } else {
-    parts.push(chrome.i18n.getMessage(
-        'readerDescriptionProductNamePartUnknown'));
-  }
-
-  if (reader.manufacturerName) {
-    parts.push(chrome.i18n.getMessage(
-        'readerDescriptionManufacturerNamePart', reader.manufacturerName));
-  }
-
-  return parts.join(' ');
-}
-
-/**
- * @param {!Array.<!chrome.usb.Device>} devices
- */
 function getUserSelectedDevicesCallback(devices) {
   logger.fine('USB selection dialog finished, ' + devices.length + ' devices ' +
               'were chosen');
-  loadDeviceList();
 }
 
 GSC.ConnectorApp.Window.DevicesDisplaying.initialize = function() {
-  loadDeviceList();
-  chrome.usb.onDeviceAdded.addListener(usbDeviceAddedListener);
-  chrome.usb.onDeviceRemoved.addListener(usbDeviceRemovedListener);
+  readerTracker = GSC.PopupWindow.Client.getData()['readerTracker'];
```

Comment:
It's similar, but not the same. Here, the word "messaging" is irrelevant. Also, allowing to pass argument as a string instead of object is strange outside the "messaging" context.

So I would agree if that would be a separate common function.